hassio/snapshots/hassio-snapshot.html

```diff
@@ -0,0 +1,260 @@
+        paper-dialog {
+          position: absolute !important;
+          bottom: 0; left: 0; right: 0;
+          border-bottom-left-radius: 0;
+          border-bottom-right-radius: 0;
+          margin: 0;
+        }
+      }
+    </style>
+    <paper-dialog id='dialog' with-backdrop on-iron-overlay-closed='dialogClosed'>
+      <div class='content'>
+        <iron-icon icon='[[computeIcon(snapshot.type)]]' class='primary'
+        ></iron-icon>
+        <div class='snapshot'>
+          [[computeName(snapshot)]]
+          <div class='snapshot-type'>
+            [[computeType(snapshot.type)]] ([[computeSize(snapshot.size)]])<br/>
+            [[formatDatetime(snapshot.date)]]
+          </div>
+        </div>
+        <div>
+          Home Assistant:
+          <paper-checkbox checked='{{restoreHass}}'>
+            Home Assistant [[snapshot.homeassistant]]
+          </paper-checkbox>
+        </div>
+        <div>
+          Folders:
+          <template is='dom-repeat' items='[[snapshot.folders]]'>
+            <paper-checkbox checked='{{item.checked}}'>
+              [[item.name]]
+            </paper-checkbox>
+          </template>
+        </div>
+        <template is='dom-if' if='[[showAddons(snapshot.addons)]]'>
+          <div>
+            Add-ons:
+            <template is='dom-repeat' items='[[snapshot.addons]]' sort='sortAddons'>
+              <paper-checkbox checked='{{item.checked}}'>
+                [[item.name]]
+                <span class='addon-version'>([[item.version]])</span>
+              </paper-checkbox>
+            </template>
+          </div>
+        </template>
+        <template is='dom-if' if='[[error]]'>
+          <p class='error'>Error: [[error]]</p>
+        </template>
+      </div>
+      <div class='card-actions'>
+        <paper-button class='warning' on-tap='deleteTapped'>Delete</paper-button>
+        <paper-button on-tap='partialRestoreTapped'>Partial restore</paper-button>
+        <template is='dom-if' if='[[isFullSnapshot(snapshot.type)]]'>
+          <paper-button on-tap='fullRestoreTapped'>Full restore</paper-button>
+        </template>
+      </div>
+    </paper-dialog>
+  </template>
+</dom-module>
+
+<script>
+class HassioSnapshot extends window.hassMixins.EventsMixin(Polymer.Element) {
+  static get is() { return 'hassio-snapshot'; }
+
+  static get properties() {
+    return {
+      hass: Object,
+      snapshotSlug: {
+        type: String,
+        observer: 'computeSnapshot',
+      },
+      snapshot: Object,
+      restoreHass: {
+        type: Boolean,
+        value: true,
+      },
+      opened: {
+        type: Boolean,
+        observer: 'openedChanged',
+        notify: true,
+      },
+      updateOverview: {
+        type: Boolean,
+        notify: true,
+      },
+      error: String,
+    };
+  }
+
+  computeSnapshot(snapshotSlug) {
+    this.hass.callApi('get', 'hassio/snapshots/' + snapshotSlug + '/info')
+      .then(function (info) {
```

Comment:
Don't use `function() { }.bind(this)` but instead use `=>`. `=>` is a lambda function that will keep the `this` context of the context where it is defined.

```js
this.hass.callApi(…).then((info) => {

})
```